main/jlp/net/stream_client.cpp

```
@@ -223,6 +230,16 @@ void run_connection(State& st, int sock) {
                (unsigned)frame_len);
     }
 
+    // Hold the ACK while voice has the radio: with no ACK the server sends
+    // no further frame, so the mic uplink runs uncontended. The frame just
+    // decoded is already on screen; the picture simply freezes for the
+    // ~few seconds of an utterance. Poll rather than block on a notify so a
+    // stop/disconnect still tears down promptly.
+    while (st.paused.load() && st.running.load()) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)